task6/script.js

```diff
@@ -0,0 +1,427 @@
+                        }
+                        optionalFieldAmount++;
+                        break;
+                    }
+                    default:
+                        break;
+                }
+            }
+        }
+
+        return optionalFieldAmount === Object.keys(filterConfig).length;
+    }
+
+    function isValid(element, index, array) {
+        for (let property in this) {
+            if (this.hasOwnProperty(property)) {
+                if (property === 'hashTags' || property === 'likes') {
+                    //hashtag or nickname of person, who liked post
+                    for (let filterElem in this[property]) {
+                        if (!element[property].find(arrayElem => arrayElem === filterElem)) {
+                            return false;
+                        }
+                    }
+                } else {
+                    if (!element.hasOwnProperty(property) || element[property] !== this[property]) {
+                        return false;
+                    }
+                }
+            }
+        }
+        return true;
+    }
+
+    function getPosts(skip = 0, top = 10, filterConfig) {
+        posts.sort(compareDates);
+        validateFilterConfig(filterConfig);
+        return posts.filter(isValid, filterConfig);
+    }
+
+    function getPost(id) {
+        if (id && typeof id === 'string') {
+            return posts.find(item => item.id === id);
+        }
+        return undefined;
+    }
+
+    function addPost(post) {
+        if (!validatePost(post)) {
+            return false;
+        }
+        posts.push(post);
+        return true;
+    }
+
+    function validateProperties(post) {
+        if (typeof post !== 'object') {
+            return false;
+        }
+        for (let property in post) {
+            if (post.hasOwnProperty(property)) {
+                switch (property) {
+                    case 'description': {
+                        if (typeof post[property] !== 'string') {
+                            return false;
+                        }
+                        break;
+                    }
+                    case 'hashTags':
+                    case 'photoLink': {
+                        if (!Array.isArray(post[property])
+                            && !post[property].every(function (element, index, array) {
+                                return typeof element === 'string';
+                            })) {
+                            return false;
+                        }
+                        break;
+                    }
+
+                    default: {
+                        return false;
+                    }
+                }
+            }
+        }
+        return true;
+    }
+
+    function editPost(id, post) {
+        if (!validateProperties(post)) {
+            return false;
+        }
+        for (let i = 0; i < posts.length; ++i) {
+            if (posts[i].id === id) {
```

Comment:
Since object is reference type, you can use `posts.find` to find post and just change it's properties